File: Problema1/main.c
/**
 * @file main.c
 * @author Alan Davide Bovo (alandavide.bovo.stud@ispascalcomandini.it)
 * @date 2024-01-03
 * Compile with: gcc main.c -o main
 */
#include<stdio.h>
#include<stdbool.h>
#include<string.h>

const char trueSignature[31] = "c0d1c3segr3t0_4bbast4n5a_lung0\0";

bool checkSignature(char *signature){
    if(strlen(signature) != strlen(trueSignature)){ // se la lunghezza della stringa inserita è diversa da quella corretta, ritorno false
        return false;
    }
    for(int i=0; i<sizeof(signature); i++){
        if(signature[i] != trueSignature[i]){ // se un carattere della stringa inserita è diverso da quello corretto, ritorno false
            return false;
        }
    }
    return true; // se la stringa inserita è uguale a quella corretta, ritorno true
}

int main(){
    char signature[50];                         // buffer in qui memorizzo la stringa inserita dall'utente
    fputs("Insert the secret code: ", stdout);  // stampo il messaggio di richiesta
    fgets(signature, 50, stdin);                // leggo al più 50 caratteri da stdin
    if(signature[strlen(signature)-1] == '\n')  // se l'ultimo carattere è un newline, lo sostituisco con un carattere nullo
        signature[strlen(signature)-1] = '\0';
    if(checkSignature(signature))               // controllo se la stringa inserita è uguale a quella corretta
        fputs("Congrats! You have found the secret code!", stdout); // se sì, stampo il messaggio di congratulazioni
    else
        fputs("Sorry, the secret code is wrong!", stdout); // altrimenti stampo il messaggio di errore
    return 0;
}

File: Problema1/main.cs
/**
 * @file main.cs
 * @author Alan Davide Bovo (alandavide.bovo.stud@ispascalcomandini.it)
 * @date 2024-01-03
 * Compile with: mcs -out:main.exe main.cs
 */
using System;

public class MainClass{
    private const string trueSignature = "c0d1c3segr3t0_4bbast4n5a_lung0"; // firma corretta

    private static bool checkSignature(string signature){   // funzione per il controllo della firma
        if(signature.Length != trueSignature.Length){       // controllo della lunghezza della stringa
            return false;
        }
        for(int i = 0; i < signature.Length; i++){          // controllo carattere per carattere
            if(signature[i] != trueSignature[i]){           // confronto tra i caratteri
                return false;
            }
        }
        return true; // se non sono stati trovati errori, la firma è corretta
    }

    public static void Main(string[] args){
        Console.Write("Insert the secret code: ");  // stampa del messaggio di input
        string signature = Console.ReadLine();      // lettura della stringa inserita dall'utente
        if(checkSignature(signature)){              // controllo della stringa inserita
            Console.WriteLine("Congrats! You have found the secret code!"); // stampa del messaggio di successo
        }else{
            Console.WriteLine("Sorry, the secret code is wrong!");          // stampa del messaggio di errore
        }
    }
}